src/CalcViewModel/Common/NavCategory.h

```diff
@@ -112,91 +87,98 @@ namespace CalculatorApp::ViewModel
-                           ViewMode viewMode,
+                           ViewModeType viewMode,
                            bool supportsNegative,
                            bool isEnabled)
                 : m_Name(name)
                 , m_AutomationName(automationName)
                 , m_Glyph(glyph)
                 , m_AccessKey(accessKey)
                 , m_modeString(mode)
-                , m_Mode(viewMode)
+                , m_ViewMode(viewMode)
                 , m_SupportsNegative(supportsNegative)
                 , m_IsEnabled(isEnabled)
             {
             }
 
         private:
-            static bool IsModeInCategoryGroup(ViewMode mode, CategoryGroupType groupType);
+            static bool IsModeInCategoryGroup(ViewModeType mode, CategoryGroupType groupType);
 
             Platform::String ^ m_modeString;
         };
 
-        [Windows::UI::Xaml::Data::Bindable] public ref class NavCategoryGroup sealed : public Windows::UI::Xaml::Data::INotifyPropertyChanged
+        [Windows::UI::Xaml::Data::Bindable]
+        public ref class NavCategoryGroup sealed : public Windows::UI::Xaml::Data::INotifyPropertyChanged
         {
+        internal:
+            NavCategoryGroup(const NavCategoryGroupInitializer& groupInitializer);
         public:
             OBSERVABLE_OBJECT();
             OBSERVABLE_PROPERTY_R(Platform::String ^, Name);
             OBSERVABLE_PROPERTY_R(Platform::String ^, AutomationName);
             OBSERVABLE_PROPERTY_R(CategoryGroupType, GroupType);
             OBSERVABLE_PROPERTY_R(Windows::Foundation::Collections::IObservableVector<NavCategory ^> ^, Categories);
+        };
 
+        public ref class NavCategoryStates sealed
+        {
+        public:
+            static void SetCurrentUser(Windows::System::User^ user);
             static Windows::Foundation::Collections::IObservableVector<NavCategoryGroup ^> ^ CreateMenuOptions();
+            static NavCategoryGroup ^ CreateCalculatorCategoryGroup();
+            static NavCategoryGroup ^ CreateConverterCategoryGroup();
+
+            static bool IsValidViewMode(ViewMode mode);
+            static bool IsViewModeEnabled(ViewMode mode);
 
-            static NavCategoryGroup ^ CreateCalculatorCategory();
-            static NavCategoryGroup ^ CreateConverterCategory();
+            // For saving/restoring last mode used.
+            static int Serialize(ViewMode mode);
+            static ViewMode Deserialize(Platform::Object ^ obj);
 
-        private:
-            NavCategoryGroup(const NavCategoryGroupInitializer& groupInitializer);
+            // Query properties from states
+            static ViewMode GetViewModeForFriendlyName(Platform::String ^ name);
+            static Platform::String ^ GetFriendlyName(ViewMode mode);
+            static Platform::String ^ GetNameResourceKey(ViewMode mode);
+            static CategoryGroupType GetGroupType(ViewMode mode);
+
+                // GetIndex is 0-based, GetPosition is 1-based
```

Comment:
Please check the indentation of this comment.

Reply:
Fixe. Thanks.